ENIGMAsystem/SHELL/Graphics_Systems/OpenGL3/GL3shader.cpp

```diff
@@ -640,7 +640,7 @@ void glsl_program_detach(int id, int sid)
 
 void glsl_program_set(int id)
 {
-  if (enigma::bound_shader != id) {
+  if (enigma::bound_shader != unsigned(id)) {
```

Comment:
What is the value of `bound_shader` if no shader is bound? Zero, I assume? Usually we use -1, but this change implies that `bound_shader` is unsigned...

Reply:
That's correct, if you read Harri's code `bound_shader`, much like the other `bound_*` variables he uses, are referring to the OpenGL resources ids, which are in fact unsigned. I was going to change `bound_shader` to signed, but then realized it would require that I change a lot more of his logic. So I chose to fix it this way, since it will still work the way he intended in order to keep the diff small.